From 2d0b2863ff233839460e242fb052fbe55d4aac3a Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:18 -0400
Subject: [cpufreq] Acquire the lock in cpufreq_policy_restore() for reading

Message-id: <1409235879-1283-45-git-send-email-prarit@redhat.com>
Patchwork-id: 89431
O-Subject: [RHEL7.1 PATCH BZ 1134639 044/185] cpufreq: Acquire the lock in cpufreq_policy_restore() for reading
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 44871c9c7f7963f8869dd8bc9620221c9e9db153
Author: Lan Tianyu <tianyu.lan@intel.com>
Date:   Wed Sep 11 15:05:05 2013 +0800

    cpufreq: Acquire the lock in cpufreq_policy_restore() for reading

    In cpufreq_policy_restore() before system suspend policy is read from
    percpu's cpufreq_cpu_data_fallback.  It's a read operation rather
    than a write one, so take the lock for reading in there.

    Signed-off-by: Lan Tianyu <tianyu.lan@intel.com>
    Reviewed-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 90252ee..33c2754 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -906,11 +906,11 @@ static struct cpufreq_policy *cpufreq_policy_restore(unsigned int cpu)
  struct cpufreq_policy *policy;
  unsigned long flags;
 
- write_lock_irqsave(&cpufreq_driver_lock, flags);
+ read_lock_irqsave(&cpufreq_driver_lock, flags);
 
  policy = per_cpu(cpufreq_cpu_data_fallback, cpu);
 
- write_unlock_irqrestore(&cpufreq_driver_lock, flags);
+ read_unlock_irqrestore(&cpufreq_driver_lock, flags);
 
  return policy;
 }
-- 
1.7.1